tests/integration/tests/test_util/util.py

```diff
@@ -294,3 +336,95 @@ def is_valid_rfc3339(date_str):
         return True
     except ValueError:
         return False
+
+
+def tracks_least_risk(track: str, arch: str) -> str:
```

Comment:
great question.  The reason behind this is to confirm that a news snap version can be upgraded to from an older existing version.  This method is necessary to find the "old version". 

the `track_least_risk` is way of looking through the channel-map of the snapstore to find a match for a given track and arch, to determine the least risky track.  Say nothing is published to stable risk in the 1.32 track for the newly supported arch `hue64`.  `amd64` and `arm64` of course work, but we're just starting to work things out for this arch.  this method helps us find the correct revision of the snap to load on the instance for a specific arch.  It may be `1.32/edge` -- but we can confirm an upgrade from `1.32/edge` -> `/path/to/local/k8s.snap` works